crates/turborepo-lib/src/cli.rs

```diff
@@ -56,6 +56,7 @@ pub enum DryRunMode {
 pub enum EnvMode {
     Infer,
     Loose,
+    StrictIncludeFrameworkVars,
```

Comment:
The name here is a bikeshed. Note that the name introducing friction for using this is intentional.

Happy to entertain other options but this is already less aggressive than my original. 😅

Reply:
I think we should go with something a bit shorter, at least for the user-facing portions. We're intending to communicate that the env vars available will match the set of env vars that `turbo` considers to be dependencies, and that means we're going infer some vars. Some ideas:
 - inferred
 - hashed
 - hashed-only
 - auto
 - detected

I'm least wild about the "hashed" variants since we don't want to force users to think about hashing. For better or worse they already need to think about inference at least a little, so I think that's less bad. "auto" and "detected" are pretty general. After writing this out, my vote goes to "auto". It suggests that `turbo` is doing something beyond what the user has specified, and I think it also contrasts nicely with "strict": "do your best guess at what will work" vs "don't do anything, I'm going to specify everything". I'm open to other [short] suggestions though.

Reply:
After implementation I don't think we can drop `infer` so we're going to be four states:
- `--infer`: we figure it out per-task based on `turbo.json`. (collapses to `loose` or `strict-include-framework-vars`)
- `--loose`: you get everything.
- `--strict`: we include only what you tell us.
- `--strict-include-framework-vars`: we include what you tell us and the framework vars.

This primary way to get this is: "you defined `passThroughEnv` but _don't_ pass in a command line flag". This is what gets you `strict-include-framework-vars`. In general the user should _rarely_ want or need to specify this.

The primary use case for manually specifying this flag is "I want strict mode for all my tasks, and for tasks that are Next.js I want to include `NEXT_PUBLIC_*`." Packages basically _never_ change frameworks, so using this is really just a feature request for "globbing over env vars" in `turbo.json`."

The problem here is that "auto", "detect", and "infer" are all synonyms and are extremely difficult to differentiate so we need two names for things which will basically never be specified on the command line, one of which is the "wrong" solution to the problem:
- "we pull it from turbo.json"
- "strict + framework vars"

Reply:
A lot of words to say:
- I think the clarity of length is valuable. This clarity also _improves_ the run summary.
- I believe that users will rarely, if ever, actually specify it.
- The friction will push people toward doing it the way we want them to, which is more-easily shareable: in `turbo.json`, in `env` and `passThroughEnv`, using a yet-unimplemented globbing feature.